include/PriorityQueue.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <iostream>
+#include <list>
+#include <exception>
+
+template<class T>
+struct MyComparator {
+	T operator()(T a, T b) const {
+		return a - b;
+	}
+};
+
+
+template<class T>
+class PriorityQueue {
+   
+public:
+	void push(const T& t); 
+	T poll();
+
+private:
+	std::list<T>  m_queue;
+};
+
+template<class T>
+inline void PriorityQueue<T>::push(const T& t)
+{
+	auto it = m_queue.begin();
+	while (it != m_queue.end() && MyComparator<T>{}(*it, t) < 0) {

```

Comment:
You’re initializing the `Comparator` on every iteration which could hurt performance. How could you improve efficiency here?